cgsmiles/graph_layout.py

```diff
@@ -199,24 +275,78 @@ def vsepr_layout(self):
                     dist[source][dest] =  d / 2 * math.sqrt(3)
         # do we need this try except????
         try:
-            p0 = nx.fruchterman_reingold_layout(self.graph)
+            p0 = nx.fruchterman_reingold_layout(self.graph, iterations=100)
         except nx.NetworkXException:
             p0 = None
 
         pos = nx.kamada_kawai_layout(self.graph,
                                      pos=p0,
                                      weight=None,
                                      dist=dist,
-                                     scale=len(self.graph) ** 0.5)
+                                     scale=1.0)
+        avg_dist = 0
+        for edge in self.graph.edges:
+            avg_dist += np.linalg.norm(pos[edge[0]]-pos[edge[1]])
+        avg_dist = avg_dist / len(self.graph.edges)
+
+        for node in pos:
+            pos[node] *= self.default_bond / avg_dist
         return pos
 
+    def circular(self):
+        positions = generate_circle_coordinates(radius=self.default_bond*2,
+                                                num_points=len(self.graph))
+
+        pos = {}
+        start = list(self.graph.nodes)[0]
+        for idx, (node, next_node) in enumerate(nx.find_cycle(self.graph, source=start)):
+            pos[node] = positions[idx]
+        print(pos)
+        return pos
+
+    def rotate_to_diagonal(self, positions):
+        # get all the distances
+        distances = squareform(pdist(positions, 'euclidean'))
+        # find pair belonging to largest distance
+        pair = unravel_index(distances.argmax(), distances.shape)
+        # get vector
+        vector = positions[pair[0]] - positions[pair[1]]
+        # compute angle
+        if self.align_with == 'diag':
+            diagonal = np.array(self.bounding_box)
+        elif self.align_with == 'x':
+            diagonal = np.array([1,0])
+        elif self.align_with == 'y':
+            diagonal = np.array([0, 1])
+
+        #print(diagonal)
+        angle = np.arctan2(diagonal[1], diagonal[0]) - np.arctan2(vector[1], vector[0])
+        # rotate object onto diagonal
+        rotation_matrix = np.array([[np.cos(angle), -np.sin(angle)],
+                                    [np.sin(angle), np.cos(angle)]])
+
+        # Apply the rotation matrix to all positions
+        rotated_positions = np.dot(positions, rotation_matrix.T)
+        return rotated_positions
```

Comment:
Why not use the align_with_y_axis function, and then rotate 45 or 90 degrees as needed?

Reply:
because that's sort of what grew organically out of some ideas